### PR TITLE
Fix GetRelativePath with different volumes

### PR DESCRIPTION
**Pull request by BrennanConroy**
#2596 
@victorhurdugaci @davidfowl

### CODE REVIEW

File: src/Microsoft.Dnx.Runtime/NuGet/Utility/PathUtility.cs [hunk 1]
```diff
@@ -86,6 +86,12 @@ public static string GetRelativePath(string path1, string path2, char separator)
             if (Microsoft.Dnx.Runtime.RuntimeEnvironmentHelper.IsWindows)
             {
                 compare = StringComparison.OrdinalIgnoreCase;
+                // check if paths are on the same volume
+                if (string.Equals(path1.Substring(1, 1), ":") && !string.Equals(path1.Substring(0, 2), path2.Substring(0, 2), compare))
```
Comment (victorhurdugaci) [inline src/Microsoft.Dnx.Runtime/NuGet/Utility/PathUtility.cs]: possible index out of range exception

Comment (davidfowl) [inline src/Microsoft.Dnx.Runtime/NuGet/Utility/PathUtility.cs]: Is that the best way to get the drive letter?

File: test/Microsoft.Dnx.Runtime.Tests/PathUtilityFacts.cs [hunk 2]
```diff
@@ -69,7 +69,8 @@ public static IEnumerable<object[]> FolderPaths
                         new object[] { "", "C:/folder/app", "C:/folder/app" },
                         new object[] { "app/", "C:/folder/app", "C:/folder/app/" },
                         new object[] { "../diff/app/", "C:/folder/app", "C:/diff/app/" },
-                        new object[] { "../folder1/file", "C:\\folder\\file", "C:\\folder1\\file" }
+                        new object[] { "../folder1/file", "C:\\folder\\file", "C:\\folder1\\file" },
```
Comment (victorhurdugaci) [inline test/Microsoft.Dnx.Runtime.Tests/PathUtilityFacts.cs]: Can you add?
1. `"C:\\", "C:\\folder1\\file"`
2. `C:\\folder1\\file", "C:\\" `

File: src/Microsoft.Dnx.Runtime/NuGet/Utility/PathUtility.cs (hunk 1 above)
Comment (davidfowl) [inline src/Microsoft.Dnx.Runtime/NuGet/Utility/PathUtility.cs]: `Path.GetPathRoot`

**Comment by BrennanConroy** [comment]
Updated

File: test/Microsoft.Dnx.Runtime.Tests/PathUtilityFacts.cs [hunk 3]
```diff
@@ -69,7 +69,10 @@ public static IEnumerable<object[]> FolderPaths
...
+                        new object[] { "H:\\folder\\file", "C:\\folder\\file", "H:\\folder\\file" },
+                        new object[] { "folder/file", "C:\\", "C:\\folder\\file" },
+                        new object[] { "..\\", "C:\\folder\\file", "C:\\" }
```
Comment (victorhurdugaci) [inline test/Microsoft.Dnx.Runtime.Tests/PathUtilityFacts.cs]: Is the result here correct?

Comment (BrennanConroy) [inline test/Microsoft.Dnx.Runtime.Tests/PathUtilityFacts.cs]: Yes

**Comment by davidfowl** [comment]
:shipit: Once travis passes

**Comment by troydai** [comment]
:shipit:

**Pull request merged**